subworkflows/local/depths.nf

```diff
@@ -0,0 +1,65 @@
+params.mag_depths_options                           = [:]
+params.mag_depths_plot_options                      = [:]
+params.mag_depths_summary_options                   = [:]
+
+include { MAG_DEPTHS                            } from '../../modules/local/mag_depths'
+include { MAG_DEPTHS_PLOT                       } from '../../modules/local/mag_depths_plot'
+include { MAG_DEPTHS_SUMMARY                    } from '../../modules/local/mag_depths_summary'
+
+/*
+ * Get number of columns in file (first line)
+ */
+def getColNo(filename) {
+    lines  = file(filename).readLines()
+    return lines[0].split('\t').size()
+}
+
+workflow DEPTHS {
+    take:
+    bins_unbins     //channel: val(meta), [ path(bins) ]
+    depths          //channel: val(meta), path(depths)
+    reads           //channel: val(meta), path(reads)
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Compute bin depths for different samples (according to `binning_map_mode`)
+    // Create a new meta joining key first, but copy meta so that
+    // we retain the information about binners and domain classification
+    ch_depth_input = bins_unbins
+        .map { meta, bins ->
+            def meta_join = meta - meta.subMap('binner','domain')
+            [ meta_join, meta, bins ]
+        }
+        .combine( depths, by: 0 )
+        .map { meta_join, meta, bins, contig_depths_file ->
+            def meta_new = meta - meta.subMap('domain')
+            [ meta_new, bins, contig_depths_file ]
+        }
+        .transpose()
+        .groupTuple(by: [0,2])
```

Comment:
Do you think using `size` or `groupKey` will make grouping operation more efficient here?

Reply:
see https://github.com/nf-core/mag/pull/499#discussion_r1326993376